test/demos/reversediffzero.jl

```diff
@@ -0,0 +1,132 @@
+"The simplest viable reverse mode a AD, only supports `Float64`"
+module ReverseDiffZero
+using ChainRulesCore
+using Test
+
+struct Tracked <: Real
+    propagate::Function
```

Comment:
Probably this is worth making type stable because we would like the optimizer to work hard on it
```suggestion
struct Tracked{F} <: Real
    propagate::F
```